code/utils.py

```diff
@@ -0,0 +1,82 @@
+import json
+import logging
+from datetime import datetime
+from enum import Enum
+from typing import Any, Dict, List, Union
+
+import numpy as np
+import pandas as pd
+
+logger = logging.getLogger(__name__)
+
+
+def clean_dataframe_for_nwb(data: pd.DataFrame) -> pd.DataFrame:
+    """
+    Clean a pandas DataFrame to ensure compatibility with NWB format.
+
+    Parameters
+    ----------
+    data : pd.DataFrame
+        The cleaned input DataFrame for NWB compatibility
+
+    Returns
+    -------
+    pd.DataFrame
+        A cleaned DataFrame that adheres to NWB data types
+    """
+    for column in data.columns:
+        # convert to nwb allowable types
+        data[column].replace({None: np.nan}, inplace=True)
+        data[column] = data[column].apply(
+            lambda x: x.value if isinstance(x, Enum) else x
+        )
+        data[column] = data[column].apply(
+            lambda x: json.dumps(x) if isinstance(x, dict) else x
+        )
+
+    return data
+
+
+def datetime_to_iso_in_dict(
```

Comment:
Can you explain why this is needed? I believe I always use iso in everything, but i may be wrong